cmd/kpod/common.go

```diff
@@ -133,3 +136,19 @@ func validateFlags(c *cli.Context, flags []cli.Flag) error {
 	}
 	return nil
 }
+
+// getCachedCreds returns the DockerAuthConfig of a registry if stored in the authfile
+func getCachedCreds(registry string, sc *types.SystemContext) (*types.DockerAuthConfig, error) {
```

Comment:
At a higher level, why is this necessary at all? `containers/image/docker` uses `config.GetAuthentication` by default. Is something else using the data, or is this doing something differently?

Reply:
This is getting the cached credentials of `registry` and parsing it to a `types.DockerAuthConfig` so that copyOptions can use this to validate the registry when pushing or pulling to and from it.
Needed to send the creds over to GetCopyOptions.

Reply:
What does ”validate the registry” mean? AFAICT `DockerRegistryCreds` is only ever used to initialize a `types.SystemContext.DockerAuthConfig`.  Wouldn't setting `DockerAuthConfig` to `nil` and setting `AuthFilePath` end up doing the same thing?

Is there a consumer of the credentials within cri-o, or elsewhere outside of c/image, that I’m overlooking?

Reply:
If I do that, there is no way for types.SystemContext to automatically call `config.GetAuthentication`. 
All `config.GetAuthentication` does is check the three files that might hold the credentials and returns it if it is found.
Unless I am missing the fact the setting `SystemContext.Authfile` knows to call GetAuthentication?

Reply:
`c/i/docker/docker_client.go:newDockerClientFromRef` _always_ calls `config.GetAuthentication`.  There is nothing to enable.

`SystemContext.AuthFile` will, via `config.getPathToAuth`, decide which file to read; all the caller should need to do is set this (or not, and leave the library to use the default).

Reply:
oooh, whoops I totally forgot GetAuthentication was being called there - my bad, sorry.
But then I need a function to call `newDockerClientFromRef` from push and pull. Should I write a function called `GetAuth` in `docker_client.go` that takes in the registry as a param and calls `newdockerClientFromRef` with that?

Reply:
No, `newDockerClientFromRef` is the default, there’s no need to opt in (and in fact, no way to opt _out_).

Just cause `c/i/docker.Reference.NewImage{Source,Destination}` to be called as usual (e.g. by `copy.Image`).  The only thing that should need to be done is passing an appropriate `types.SystemContext` through the call stack.

Reply:
right, got it! I fixed the code accordingly, @mtrmac PTAL
Thanks for the help!